### ISSUE TITLE
Task execution role not set in new task revision

### ISSUE THREAD

**Issue by mhuebner** [silinternational/ecs-deploy#186]
When ecs-deloy creates a new revision of a taks the Task execution role is not set in the new revision of the task definition.

When i look into the ecs-deploy script it seems that the `executionRoleArn`  is only set if  Type is FARGATE and not EC2. It would be great when EC2 also had `executionRoleArn`  set for the new task def!

**Comment by mhuebner** [comment]
Line 311 to 329 could be replaced with

```
# Default JQ filter for new task definition
    NEW_DEF_JQ_FILTER="family: .family, volumes: .volumes, containerDefinitions: .containerDefinitions, placementConstraints: .placementConstraints"

    # Some options in task definition should only be included in new definition if present in
    # current definition. If found in current definition, append to JQ filter.
    CONDITIONAL_OPTIONS=(networkMode taskRoleArn placementConstraints executionRoleArn requiresCompatibilities cpu memory)
    for i in "${CONDITIONAL_OPTIONS[@]}"; do
      re=".*${i}.*"
      if [[ "$DEF" =~ $re ]]; then
        NEW_DEF_JQ_FILTER="${NEW_DEF_JQ_FILTER}, ${i}: .${i}"
      fi
    done
```

EC2 and FARGATE both have the same constraints.

**Comment by mhuebner** [comment]
Relevant Specs must be refactored as well

**Comment by theintz** [comment]
This PR fixes the issue: https://github.com/silinternational/ecs-deploy/pull/179

**Comment by mhuebner** [comment]
Yeah, but merge seems to be blocked.

**Comment by olivierto** [comment]
+1 my CI is broken now that i'm using private registry auth in my task definitions :'(

**Comment by abaranovv** [comment]
+1 same case for private registry auth

**Comment by fillup** [comment]
This has been fixed and released as version 3.7.0.
